services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/QinQFlowSpec.groovy

```diff
@@ -0,0 +1,365 @@
+        def swP = topologyHelper.getAllNeighboringSwitchPairs().find {
+            [it.src, it.dst].any { northbound.getSwitchProperties(it.dpId).multiTable }
+        } ?: assumeTrue("No suiting switches found", false)
+
+        def flow = flowHelperV2.randomFlow(swP)
+        flow.source.innerVlanId = srcInnerVlanId
+        flow.destination.innerVlanId = dstInnerVlanId
+        northboundV2.addFlow(flow)
+
+        then: "Human readable error is returned"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 404
+        //TODO(andriidovhan) add error message
+        exc.responseBodyAsString.to(MessageError).errorMessage == "Could not create flow"
+
+        cleanup:
+        flow && flowHelperV2.deleteFlow(flow.flowId)
+
+        where:
+        srcInnerVlanId | dstInnerVlanId | errorMsg
+        4096           | 10             | ""
+        10             | -1             | ""
+    }
+
+    @Ignore("is not implemented yet")
+    def "System doesn't allow to create/update/delete a QinQ flow via APIv1"() {
+        given: "Two switches with enabled multi table mode"
+        def swP = topologyHelper.getAllNeighboringSwitchPairs().find {
+            [it.src, it.dst].every { northbound.getSwitchProperties(it.dpId).multiTable }
+        } ?: assumeTrue("No suiting switches found", false)
+
+        when: "Create a QinQ flow via APIv1"
+        def flowV1 = flowHelper.randomFlow(swP)
+        flowV1.source.innerVlanId = 100
+        flowV1.destination.innerVlanId = 200
+        northbound.addFlow(flowV1)
+
+        then: "Human readable error is returned"
+        def eCreate = thrown(HttpClientErrorException)
+        eCreate.rawStatusCode == 400
+        //TODO(andriidovhan) add error message
+        eCreate.responseBodyAsString.to(MessageError).errorMessage == "Could not create the flow"
+
+        when: "Create a QinQ flow via APIv2"
+        def flowV2 = flowHelperV2.randomFlow(swP)
+        flowV2.source.innerVlanId = 234
+        flowV2.destination.innerVlanId = 432
+        northboundV2.addFlow(flowV2)
+
+        and: "Try to update the flow via APIv1"
+        flowHelperV2.updateFlow(flowV2.flowId, flowV2.tap { flowV2.source.innerVlanId += 1 })
+
+        then: "Human readable error is returned"
+        def eUpdate = thrown(HttpClientErrorException)
+        eUpdate.rawStatusCode == 400
+        //TODO(andriidovhan) add error message
+        eUpdate.responseBodyAsString.to(MessageError).errorMessage == "Could not update the flow"
+
+        when: "Try to delete the flow via APIv1"
+        northbound.deleteFlow(flowV2.flowId)
+
+        then: "Human readable error is returned"
+        def eDelete = thrown(HttpClientErrorException)
+        eDelete.rawStatusCode == 400
+        //TODO(andriidovhan) add error message
+        eDelete.responseBodyAsString.to(MessageError).errorMessage == "Could not delete flow"
+
+        cleanup:
+        flowV2 && flowHelperV2.deleteFlow(flowV2.flowId)
+    }
+}
```

Comment:
I'm wondering what will happen if I don't set inner/outer vlans at all (leave 'null' instead of 0)

Reply:
default value for inner vlan is `0`